txnsync/peerscheduler.go

```diff
@@ -55,9 +97,32 @@ func (p *peerScheduler) Len() int {
 	return len(p.peers)
 }
 
+func (p *peerScheduler) replaceIndices(indices []int, i, j int) {
+
+	for idx, x := range indices {
+		if x == i {
+			indices[idx] = j
+		} else if x == j {
+			indices[idx] = i
+		}
+	}
```

Comment:
In this loop you're doing 2 comparison per element.. which is not needed.
The indices array is already ordered, so why not to take advantage of that ?
i.e. I can think of -
1. binary search
2. sort I & j. Scan to find x <= i first, then do the same for j.

Reply:
The problem is that the array `indices` is sorted by the `p.peers[indices[i]].next` value. So binary search will not be straightforward. 
Given this, and knowing that this array is not going to be more than several elements long, two checks should not be a problem.